axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetLineMoveServiceImpl.java

```diff
@@ -324,7 +325,8 @@ protected Move generateImpairementAccountMove(FixedAssetLine fixedAssetLine, boo
         Account debitLineAccount;
         Account creditLineAccount;
         BigDecimal amount;
-        if (impairmentValue.compareTo(BigDecimal.ZERO) > 0) {
+        if (impairmentValue.signum() > 0 && accountingValue.signum() > 0
```

Comment:
can you add some brackets to be sure the expression is correctly evaluated ? And to make it more understandable.